
enum EValue {
    V_0 = 0;
    V_1 = 1;
    V_5 = 5;
    V_1000 = 1000;
};

message TBasicNode {
    repeated TBasicNode Nodes = 1;
}

message TNode {
    optional float Value = 1;
    repeated TNode Nodes = 2;
    optional TNode OneNode = 3;
};

message TValues {
    optional string ValString = 1;
    optional int64 ValInt64 = 2;
    required int32 ValInt32 = 3 [default = 1917];
    required uint64 ValUInt64 = 4 [default = 1789];
    repeated uint32 ValUInt32 = 5;
    repeated bool ValBool = 6;
    optional float ValFloat = 7;
    optional double ValDouble = 8;
    optional EValue ValEnum = 9;
};

message TOneInt64 {
    optional int64 Value = 1;
};

message TOneInt32 {
    optional int32 Value = 1;
};

message TOneUInt64 {
    optional uint64 Value = 1;
};

message TOneUInt32 {
    optional uint32 Value = 1;
};

message TOneDouble {
    optional double Value = 1;
};

message TOneFloat {
    optional float Value = 1;
};

message TOneString {
    optional string Value = 1;
};

message TOneEnum {
    optional EValue Value = 1;
};

message TEmpty {
};

